src/vma/dev/net_device_val.cpp

```diff
@@ -743,7 +743,11 @@ int net_device_val::global_ring_request_notification(uint64_t poll_sn)
 	int ret_total = 0;
 	auto_unlocker lock(m_lock);
 	rings_hash_map_t::iterator ring_iter;
+	int rhm_sz = m_h_ring_map.size();
 	for (ring_iter = m_h_ring_map.begin(); ring_iter != m_h_ring_map.end(); ring_iter++) {
+		if (rhm_sz != (int)m_h_ring_map.size()) {
+			break;
+		}
```

Comment:
so this change stops processing in case `m_h_ring_map` size is changed. May be log_dbg() or log_error() depending on such case valid/error is good to be added before break?